components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/internal/OpenIDConnectServiceComponent.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.openidconnect.internal;
+
+import edu.emory.mathcs.backport.java.util.Collections;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.core.util.IdentityCoreInitializedEvent;
+import org.wso2.carbon.identity.openidconnect.OpenIDConnectClaimFilter;
+
+import java.util.Comparator;
+
+@Component(
+        name = "identity.openidconnect.component",
+        immediate = true
+)
+public class OpenIDConnectServiceComponent {
+
+    private Log log = LogFactory.getLog(OpenIDConnectServiceComponent.class);
+
+    /**
+     * Set {@link OpenIDConnectClaimFilter} implementation
+     *
+     * @param openIDConnectClaimFilter an implementation of {@link OpenIDConnectClaimFilter}
+     */
+    @Reference(
+            name = "openid.connect.claim.filter.service",
+            service = OpenIDConnectClaimFilter.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetOpenIDConnectClaimFilter"
+    )
+    protected void setOpenIDConnectClaimFilter(OpenIDConnectClaimFilter openIDConnectClaimFilter) {
+        if (log.isDebugEnabled()) {
+            log.debug("OpenIDConnectClaimFilter: " + openIDConnectClaimFilter.getClass().getName() + " set in " +
+                    "OpenIDConnectServiceComponent.");
+        }
+        OpenIDConnectServiceComponentHolder.getInstance().addOpenIDConnectClaimFilter(openIDConnectClaimFilter);
+        Collections.sort(OpenIDConnectServiceComponentHolder.getInstance().getOpenIDConnectClaimFilters(),
```

Comment:
Sorting the collection outside of the OpenIDConnectServiceComponentHolder breaks the proper OOP encapsulation principle. But anyway OpenIDConnectServiceComponentHolder is not OOP.

Since we do this here, there is no need of addOpenIDConnectClaimFilter() or removeOpenIDConnectClaimFilter().